### DISCUSSION TITLE
[QUESTION] Why exactly do I need agents ?

### DISCUSSION THREAD

**Discussion by esakarya** [rancher/rke2#3200] [General]
I have completed the setup of the cluster consisting of 3 servers.
I have not added any agent servers.
I'm about to get into production, but before that, what you need to learn is **why exactly do I need agent servers?**
**Agents need to be higher than the processor and ram server servers ?**

**Comment by brandond** [comment]
It's partially a matter of preference and partially a question of scaling your cluster. You don't need more than 3-5 nodes running etcd. You don't really need more than 2 nodes running the control-plane components. If you run control-plane+etcd, this means that 3 server nodes is fine, if you're OK with your workload also running on those same nodes.

When you start scaling out, and are worried about the workload competing with etcd and the control-plane for host resources, it makes sense to add agents, and taint the server nodes so that your workloads don't run there.
